### ISSUE TITLE
meld does not work

### ISSUE THREAD

**Issue by wwat** [NixOS/nixpkgs#6514]
```
GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.

(.meld-wrapped:979): Gtk-WARNING **: Could not find the icon 'drive-optical'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
        http://icon-theme.freedesktop.org/releases
Traceback (most recent call last):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/diffmap.py", line 130, in do_draw
    width, height)
cairo.Error: invalid value (typically too big) for the size of the input (surface, pattern, etc.)
Traceback (most recent call last):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/diffmap.py", line 130, in do_draw
    width, height)
cairo.Error: invalid value (typically too big) for the size of the input (surface, pattern, etc.)
Traceback (most recent call last):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 616, in <lambda>
    self.append_diff(arg, **kwargs))
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 678, in append_diff
    paths, merge_output=merge_output, meta=meta)
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 645, in append_filediff
    doc = filediff.FileDiff(len(files))
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/filediff.py", line 362, in __init__
    from meld.gutterrendererchunk import GutterRendererChunkAction
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 33, in <module>
    class GutterRendererChunkAction(GtkSource.GutterRendererPixbuf):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 38, in GutterRendererChunkAction
    MODE_REPLACE: load("meld-change-apply-right"),
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 30, in load
    return icon_theme.load_icon(icon_name, LINE_HEIGHT, 0)
gi._glib.GError: Icon 'meld-change-apply-right' not present in theme
Traceback (most recent call last):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 616, in <lambda>
    self.append_diff(arg, **kwargs))
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 678, in append_diff
    paths, merge_output=merge_output, meta=meta)
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/meldwindow.py", line 645, in append_filediff
    doc = filediff.FileDiff(len(files))
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/filediff.py", line 362, in __init__
    from meld.gutterrendererchunk import GutterRendererChunkAction
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 33, in <module>
    class GutterRendererChunkAction(GtkSource.GutterRendererPixbuf):
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 38, in GutterRendererChunkAction
    MODE_REPLACE: load("meld-change-apply-right"),
  File "/nix/store/1l57xj9mi30a5dd9zbzdqsrqr3l7ckar-meld-3.12.3/lib/python2.7/site-packages/meld/gutterrendererchunk.py", line 30, in load
    return icon_theme.load_icon(icon_name, LINE_HEIGHT, 0)
gi._glib.GError: Icon 'meld-change-apply-right' not present in theme
```
This is the same bug as in reported here: https://bugzilla.redhat.com/show_bug.cgi?id=1108079 (which you can consult for more information)

**Comment by bjornfor** [comment]
Meld works for me, but I have it installed in systemPackages along with many other packages (on NixOS).

How did you install and run it? Running NixOS or just nix?

**Comment by vcunat** [comment]
I manage to make it miss some icons by unsetting ``$XDG_DATA_DIRS``, but it just runs without showing a few icons.

**Comment by wwat** [comment]
This is running on Ubuntu. 

It "runs" fine, until the point that I try to compare two directories and then press on a file where there is a difference. At that point a file is loaded, and that file is not there, and that's counted as a fatal error.  (Typical of a hacked up programming language like Python.)

The purpose of the program meld is to compare differences. A release of such a program should come with tests. Additionally, it shouldn't depend on the specific theme one is running, because that's just... retarded. Now, I am not saying that this is a Nix issue per se, but reconsidering the minimum quality bar for a release is a thing.

**Comment by bobvanderlinden** [comment]
Hmm, works here on nixos and archlinux. Comparing directories and opening files.
The change that Redhat did is shown here: http://pkgs.fedoraproject.org/cgit/meld.git/commit/?h=master&id=7cb29ea3b986bb683a0c252dae8e9000032b9ee9
It has something to do with `gtk-update-icon-cache`?

Other than that, I see the first line shows that `GIO_EXTRA_MODULES` isn't set properly. I had the same issue with gnumeric and fixed it in #6331.

**Comment by vbgl** [comment]
Fixed with commit 9d5508d85c33b8fb22d79dde6176792eac2c2696.
